apps/demos/src/app/rx-angular-pocs/template/directives/for/rx-for.directive.ts

```diff
@@ -22,6 +22,175 @@ import {
+ * `ChangeDetection` for `projected views` (aka `@ContentChild` or `@ViewChild`).
+ *
+ * Read more about this in the example section
+ *
+ * ### Context Variables
+ *
+ * The following context variables are available for each template:
+ *
+ * - $implicit: `T`
+ * - item$: `Observable<T>`
+ * - index: `number`
+ * - count: `number`
+ * - first: `boolean`
+ * - last: `boolean`
+ * - even: `boolean`
+ * - odd: `boolean`
+ * - index$: `Observable<number>`
+ * - count$: `Observable<number>`
+ * - first$: `Observable<boolean>`
+ * - last$: `Observable<boolean>`
+ * - even$: `Observable<boolean>`
+ * - odd$: `Observable<boolean>`
+ * - select: (keys: string[], distinctByMap): Observable<Partial<T>> (reactivity in the embedded view)
+ *
+ * ### Input properties
+ *
+ *  - trackBy: `(index: number, item: T) => any`
+ *  - trackBy: `keyof T`
+ *  - distinctBy: `(item: T, item: T) => boolean`
+ *  - strategy: `string`
+ *  - strategy: `Observable<string>`
+ *  - parent: boolean;
+ *  - renderCallback: Subject<T[]>
+ *
+ *
+ * ### Features of `*rxFor`
+ *
+ * Included features for `*rxFor`:
+ * - Push based architecture
+ * - Immutable as well as mutable data structures (with `trackBy` & `distinctBy`)
+ * - Control `updates` via `distinctBy` and `moves` via `trackBy`
+ * - Reactive as well as imperative values in the template (`ngFor` drop-in replacement)
+ * - `ListManager`: special logic for differ mechanism to avoid over-rendering; abstracts away low level logic
+ * - render every EmbeddedView on its own while applying the configured StrategyCredentials#behavior
+ * - detach EmbeddedView to avoid ChangeDetection top-down rendering interference when scheduling
+ * - cancel any scheduled work if a remove was triggered for an `trackById` / `distinctById`
+ * - cancel an update if a new update was triggered for the same `trackById` / `distinctById`
+ *
+ *
+ * ### Simple example using `*rxFor` with `Observable` values
+ * ```html
+ * <ul>
+ *   <li *rxFor="let item of observableItems$; trackBy: trackItem">
+ *      {{ item }}
+ *   </li>
+ * </ul>
+ * ```
+ *
+ * ### Simple example using `*rxFor` with simple static values
+ * ```html
+ * <ul>
+ *   <li *rxFor="let item of items; trackBy: trackItem">
+ *      {{ item }}
+ *   </li>
+ * </ul>
+ * ```
+ *
+ *
+ * ### Using the context variables
+ *
+ * ```html
+ * <ul>
+ *   <li
+ *     *rxFor="
+ *       let item of observableItems$;
+ *       let count = count;
+ *       let index = index;
+ *       let first = first;
+ *       let last = last;
+ *       let even = even;
+ *       let odd = odd;
+ *       trackBy: trackItem;
+ *       strategy: 'normal';
+ *       parent: true;
+ *     "
+ *   >
+ *     <div>{{ count }}</div>
+ *     <div>{{ index }}</div>
+ *     <div>{{ item }}</div>
+ *     <div>{{ first }}</div>
+ *     <div>{{ last }}</div>
+ *     <div>{{ even }}</div>
+ *     <div>{{ odd }}</div>
+ *   </li>
+ * </ul>
+ * ```
+ *
```

Comment:
Could we do an issue for `distinctBy` and to and explain why it can help?

Reply:
i tried to explain it in a simple sentence that developers can control updates via distinctBy and moves via trackBy. But I think its not very clear